Builds/CMake/CMakeFuncs.cmake

```diff
@@ -457,6 +457,9 @@ macro(setup_build_boilerplate)
       add_definitions(-DSANITIZER=TSAN)
     endif()
   endif()
+  if (NOT WIN32)
+    add_compile_options("-pthread")
```

Comment:
@mellery451 Good suggestions. Done.

Bike shed: We may want to rename the `find_*` functions to `use_*`. Rational: A function that "finds" something should be side-effect free (other than setting results variables for the various directories and flags), it shouldn't change any variables that  effect the target. That's the reason I didn't put it in `find_pthread` originally. Looking at `find_boost` however shows we're already doing this. I don't want to do the rename for this PR, but maybe in a later re-factor.